addons/api/addon/models/role.js

```diff
@@ -42,6 +51,61 @@ export default class RoleModel extends GeneratedRoleModel {
       : null;
   }
 
+  /**
+   * Convenience for looking up the grant scopes, if loaded.
+   */
+  get grantScopes() {
+    // filter out scopes that start with an o_ or p_
+    // then sort the keywords
+    const grantScopes = [];
+    if (this.grant_scope_ids) {
+      const keywordIDs = this.grant_scope_ids.filter((id) => {
+        return !id.startsWith('o_') && !id.startsWith('p_') && id !== 'global';
+      });
+      // using filter instead of find to get the global id and allow for
+      // spreading into sortedScopeIDs if globalID is undefined
+      const globalID = this.grant_scope_ids.filter((id) => {
+        return id === 'global';
+      });
+      const orgIDs = this.grant_scope_ids.filter((id) => {
+        return id.startsWith('o_');
+      });
+      const projectIDs = this.grant_scope_ids.filter((id) => {
+        return id.startsWith('p_');
+      });
```

Comment:
I don't have much context here- do we want to sort the project and org here? In some resources, we display global first and display the rest of the scopes after global and don't sort them so wondering if this is needed

Reply:
In the original designs we didn't need this but when reading all of the grant scopes on a role, we wanted a way to view the keywords (this, children, descendants), orgs, and projects in the same table. With that, we decided to have the keywords first, global second, orgs third, and projects last. We wanted global and orgs to be at the top and not mixed in among the projects.